src/orders/service/order.service.ts

```diff
@@ -0,0 +1,265 @@
+      await queryRunner.rollbackTransaction();
+      throw error;
+    } finally {
+      await queryRunner.release();
+    }
+  }
+
+  async deleteOrder(orderId: string): Promise<void> {
+    const queryRunner = this.dataSource.createQueryRunner();
+    await queryRunner.connect();
+    await queryRunner.startTransaction();
+
+    try {
+      const existingOrder = await this.orderRepository.findById(orderId);
+      if (!existingOrder) {
+        throw new NotFoundException('No order found by given ID');
+      }
+
+      const existingOrderDetails = existingOrder.orderDetails;
+      await this.restoreStock(existingOrderDetails);
+
+      await queryRunner.manager.delete(OrderDetail, { order: { id: orderId } });
+      await queryRunner.manager.delete(Order, orderId);
+
+      await queryRunner.commitTransaction();
+    } catch (error) {
+      await queryRunner.rollbackTransaction();
+      throw error;
+    } finally {
+      await queryRunner.release();
+    }
+  }
+
+  private async restoreStock(orderDetails: OrderDetail[]) {
+    for (const detail of orderDetails) {
+      const stock = await this.stockService.getByProductIdAndLocationId(
+        detail.product.id,
+        detail.shippedFrom.id,
+      );
+      await this.stockService.update(
+        stock.id,
+        stock.quantity + detail.quantity,
+      );
+    }
+  }
+
+  private async getOrderedItemsStocks(
+    desiredOrderItems: DesiredOrderItem[],
+  ): Promise<Stock[]> {
+    return await this.stockService.getStocksByProductsAndLocations(
+      desiredOrderItems,
+    );
+  }
+
+  private validateStockAvailability(
+    orderedItemsStocks: any[],
+    desiredOrderItems: DesiredOrderItem[],
+  ): void {
+    const insufficientStockItems = [];
+    for (const item of desiredOrderItems) {
+      const stock = orderedItemsStocks.find(
+        (stock) =>
+          stock.product.id === item.productId &&
+          stock.location.id === item.locationId,
+      );
+
+      if (!stock || stock.quantity < item.desiredQuantity) {
+        insufficientStockItems.push({
+          productId: item.productId,
+          locationId: item.locationId,
+          availableQuantity: stock ? stock.quantity : 0,
+          desiredQuantity: item.desiredQuantity,
+        });
+      }
+    }
+
+    if (insufficientStockItems.length > 0) {
+      throw new BadRequestException({
+        message: 'Insufficient stock for some items',
+        insufficientStockItems,
+      });
+    }
+  }
+
+  private createOrderEntity(
+    newOrder: Omit<Order, 'id' | 'orderDetails' | 'customer' | 'createdAt'>,
+    customer: any,
+  ): Order {
+    const order = new Order();
+    order.customer = customer;
+    order.country = newOrder.country;
+    order.city = newOrder.city;
+    order.county = newOrder.county;
+    order.streetAddress = newOrder.streetAddress;
+    order.createdAt = new Date();
+    return order;
+  }
+
+  private createOrderDetails(
+    order: Order,
+    orderedItemsStocks: any[],
```

Comment:
**issue:** please avoid the `any` type in all circumstances, use a specific type

Reply:
Big mistake on my side. I was debugging and then forgot to change these back. Im onto it!